Assets/Scripts/Goal.cs

```diff
@@ -151,6 +177,32 @@ enum GoalState {
         goalState = stateToSwitch;
     }
 
+    [PunRPC]
+    private void Goal_SetMaterial(byte stateToSwitch, bool  isHit) {
+        if (isHit) {
+            StartCoroutine(ShowGoalHitFor(0.5f));
+        } else {
+            GoalState state = (GoalState)stateToSwitch;
+            if (state == GoalState.STATIONARY) {
+                SwitchGoalMaterial(stationaryMat);
+            } else if (state == GoalState.FOLLOWING) {
+                SwitchGoalMaterial(followingMat);
```

Comment:
does code execution ever reach this path? it looks like the only times this function is called are with `GoalState.FOLLOWING` and `isHit = false`, and with `GoalState.STATIONARY` and `isHit = true`

Reply:
this is for whenever a goal is scored

Reply:
much as that makes sense logically, i don't think this code block is the one doing that. when you call this function with `((isHit==true) && GoalState.STATIONARY)` it goes into the `if (state == GoalState.STATIONARY)` path and exits this function after it calls `SwitchGoalMaterial(stationaryMat);`

searching the entire project for where this function is used shows that it is only ever called with either `((isHit==true) && GoalState.STATIONARY)` or `((isHit==false) && GoalState.FOLLOWING)`, but never `((isHit==true)&&GoalState.FOLLOWING)`, hence this path of code execution is never taken

(note that `state` in this function only directly refers to the argument passed in, not the `goalState` field of the class)

Reply:
noted, I'll remove it